s2spy/rgdr/rgdr.py

```diff
@@ -355,91 +355,132 @@ def __init__(
                 smaller than this minimum area will be discarded.
 
         Attributes:
-            corr_map (float): correlation coefficient map of given precursor field and
+            corr_map: correlation coefficient map of given precursor field and
                 target series.
-            pval_map (float): p-values map of correlation
-            cluster_map (U32): cluster labels for precursor field masked by p-values
+            pval_map: p-values map of correlation
+            cluster_map: cluster labels for precursor field masked by p-values
         """
-        self.corr_map: xr.DataArray
-        self.pval_map: xr.DataArray
-        self.cluster_map = None
-        self._area = None
+        self._lag = lag
         self._dbscan_params = {"eps": eps_km, "alpha": alpha, "min_area": min_area_km2}
 
+        self._target_intervals = (
+            [target_intervals]
+            if isinstance(target_intervals, int)
+            else target_intervals
+        )
+        self._precursor_intervals = utils.intervals_subtract(
+            self._target_intervals, lag
+        )
+
+        self._corr_map: Union[None, xr.DataArray] = None
+        self._pval_map: Union[None, xr.DataArray] = None
+        self._cluster_map: Union[None, xr.DataArray] = None
+
+        self._area = None
+
+    @property
+    def target_intervals(self) -> List[int]:
```

Comment:
For my own learning experience, can I ask what the added value of this function is? self._target_intervals is already a list (defined at __init__), so why create this function and not simply name:

```
        self.target_intervals = (
             [target_intervals]
             if isinstance(target_intervals, int)
             else target_intervals
         )
```

Reply:
Maybe it's for consistency reasons? Later on I see the added value of calling the method cluster map for example.

Reply:
The `@property` decorator is so we can give users information without allowing them to change anything. It is not a function in the way that users call it like a class property, e.g.:
```py
rgdr.target_intervals
```
And not:
```py
rgdr.target_intervals()  # raises an error
```

We do not want to allow users to change properties of the RGDR class after it is initiated. The underscore at the beginning of the variable name `self._target_intervals` denotes that users _should not_ touch this, and if they do so anyway that they're responsible for any weird behavior.

If we do not implement `@property`, a user would be able to do:
```py
rgdr.target_intervals = 5
```
And the lag & precursor intervals would not be affected.

If it is a `@property`, and no `@target_intervals.setter` has been implemented, Python will raise an error when a user tries to modify the value.

This is also the reason why I made the stored maps properties.